### PR TITLE
Best Practices Updates

### PR DESCRIPTION
**Pull request by allierowan**
updating this project so that it follows some engineering best practices:

- [x] add PR template
- [x] make sure tests are passing on master
- [x] enable code coverage for the js

cc @miccolis @karenzshea

### CODE REVIEW

File: test/basic/extractor.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ BOOST_AUTO_TEST_CASE(annotator_test_basic)
 
     Database db(true);
     Extractor extractor(buffer.c_str(), buffer.size(), "xml", db);
+    db.build_rtree();
```
Comment (danpat) [inline test/basic/extractor.cpp]: @allierowan Why this change?  The `Database db(true);` line above should signal the `Extractor` to build the RTree - if it's not, then this test is possibly failing legitimately.

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+What issue is this PR targeting? In a few words, what does the PR add, change, or remove?
+
+## Tasklist
+
+- [ ] Write code & tests
+- [ ] Update relevant CHANGELOG and/or the README
+- [ ] Get a review
+- [ ] Rebase and clean up commits
+- [ ] Merge & release
```
Comment (miccolis) [inline PULL_REQUEST_TEMPLATE.md]: Are there docs about what releasing looks like?

Comment (allierowan) [reply PULL_REQUEST_TEMPLATE.md]: there is a fairly detailed section in the README about releasing - I can add a note here.

Comment (miccolis) [reply PULL_REQUEST_TEMPLATE.md]: oh, if it's there that's fine... took a glance at the readme but apparently not a good enough one. :shipit:

**Pull request merged**